iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/resource/tsfile/PipeTsFileResourceManager.java

```diff
@@ -132,42 +131,53 @@ private void ttlCheck() {
   public File increaseFileReference(
       final File file, final boolean isTsFile, final TsFileResource tsFileResource)
       throws IOException {
-    lock.lock();
+    // If the file is already a hardlink or copied file,
+    // just increase reference count and return it
+    segmentLock.lock(file);
     try {
-      // If the file is already a hardlink or copied file,
-      // just increase reference count and return it
-      if (increaseReferenceIfExists(file.getPath())) {
+      if (increaseReferenceIfExists(file)) {
         return file;
       }
+    } finally {
+      segmentLock.unlock(file);
+    }
 
-      // If the file is not a hardlink or copied file, check if there is a related hardlink or
-      // copied file in pipe dir. if so, increase reference count and return it
-      final File hardlinkOrCopiedFile = getHardlinkOrCopiedFileInPipeDir(file);
-      if (increaseReferenceIfExists(hardlinkOrCopiedFile.getPath())) {
+    // If the file is not a hardlink or copied file, check if there is a related hardlink or
+    // copied file in pipe dir. if so, increase reference count and return it
+    final File hardlinkOrCopiedFile = getHardlinkOrCopiedFileInPipeDir(file);
+    segmentLock.lock(hardlinkOrCopiedFile);
+    try {
+      if (increaseReferenceIfExists(hardlinkOrCopiedFile)) {
```

Comment:
I think `getHardlinkOrCopiedFileInPipeDir(file)` and `segmentLock.lock(hardlinkOrCopiedFile)`; this part of the code can be completed in the `file lock`. Similarly, `resultFile` is obtained in the `hardlinkOrCopiedFile lock`, which can better reduce lock contention.